src/inc/RingBuffer.h

```diff
@@ -6,146 +6,150 @@
+        // Compute weighted sum of samples before and after fractional index.
+        float frac = fracIndex - index;
+        return prevSample * (1 - frac) + nextSample * frac;
     }
 
     /*! set buffer content and indices to 0
     \return void
     */
-    void reset()
-    {
-        std::memset (m_ptBuff, 0, sizeof(T)*m_iBuffLength);
-        m_iReadIdx  = 0;
-        m_iWriteIdx = 0;
+    void reset() {
+        std::memset(buffer, 0, sizeof(T) * length);
+        head = 0;
+        tail = 0;
     }
 
     /*! return the current index for writing/put
     \return int
     */
-    int getWriteIdx() const
-    {
-        return m_iWriteIdx;
+    int getWriteIdx() const {
+        return head;
     }
 
     /*! move the write index to a new position
     \param iNewWriteIdx: new position
     \return void
     */
-    void setWriteIdx(int iNewWriteIdx)
-    {
-        incIdx(m_iWriteIdx, iNewWriteIdx - m_iWriteIdx);
+    void setWriteIdx(int iNewWriteIdx) {
+        head = mod(iNewWriteIdx, length);
     }
 
     /*! return the current index for reading/get
     \return int
     */
-    int getReadIdx() const
-    {
-        return m_iReadIdx;
+    int getReadIdx() const {
+        return tail;
     }
 
     /*! move the read index to a new position
     \param iNewReadIdx: new position
     \return void
     */
-    void setReadIdx(int iNewReadIdx)
-    {
-        incIdx(m_iReadIdx, iNewReadIdx - m_iReadIdx);
+    void setReadIdx(int iNewReadIdx) {
+        tail = mod(iNewReadIdx, length);
     }
 
     /*! returns the number of values currently buffered (note: 0 could also mean the buffer is full!)
     \return int
     */
-    int getNumValuesInBuffer() const
-    {
-        return (m_iWriteIdx - m_iReadIdx + m_iBuffLength) % m_iBuffLength;
+    int getNumValuesInBuffer() const {
+        // If `tail > head`, we assume it's because `head` has wrapped around.
+        return tail > head ? (head + length - tail) : (head - tail);
     }
 
     /*! returns the length of the internal buffer
     \return int
     */
-    int getLength() const
-    {
-        return m_iBuffLength;
+    int getLength() const {
+        return length;
     }
 private:
-    CRingBuffer();
-    CRingBuffer(const CRingBuffer& that);
-
-    void incIdx(int& iIdx, int iOffset = 1)
-    {
-        while ((iIdx + iOffset) < 0)
-        {
-            // avoid negative buffer indices
-            iOffset += m_iBuffLength;
-        }
-        iIdx = (iIdx + iOffset) % m_iBuffLength;
-    };
-
-    int m_iBuffLength = 0,      //!< length of the internal buffer
-        m_iReadIdx = 0,         //!< current read index
-        m_iWriteIdx = 0;        //!< current write index
-
-    T* m_ptBuff = 0;            //!< data buffer
+    RingBuffer();
+    RingBuffer(const RingBuffer& that);
+
+    // Perform post-increment with wrapping.
+    int postInc(int &index) {
+       const int value = index++;
+       if (index == length) index = 0;
+       return value; 
+    }
```

Comment:
This worries me slightly about no checking if index > length, is there a guarantee somewhere else that index starts less than length? Otherwise maybe a while loop of while (index >= length) index -= length